src/workloads/c2c/MongosyncScripts.yml

```diff
@@ -0,0 +1,95 @@
+SchemaVersion: 2018-07-01
+
+# This is a collection of helper scripts used to communicate with mongosync. To use them,
+# use LoadConfig and load the script you would like to use in your actor's phase.
+# For example:
+# - Name: StartMongosync
+#   Type: ExternalScriptRunner
+#   Threads: 1
+#   Phases:
+#     - LoadConfig:
+#       Path: ./MongosyncScripts.yml
+#       Key: StartMongosync
+#
+# Note that the polling scripts start with set -eou pipefail so that any
+# subcommand run during the script failing will cause the entire script to fail.
+# This way if mongosync crashes during the test, it will stop the test execution
+# immediately and genny will bubble up the error and fail the test
+#
+# For curl commands we pass -s and -S so that an errors are still reported (-S)
+# but download progress stats are not logged to stderr
+#
+StartMongosync:
+  Repeat: 1
+  Command: sh
+  MetricsName: StartMongosync
+  Script: |
+    curl -s -S -X POST http://10.2.0.160:27182/api/v1/start \
+      -H 'Content-Type: application/json' \
+      -d '{"Source": "cluster0", "Destination": "cluster1"}'
+
+PollForCEA:
+  Repeat: 1
+  Command: sh
+  MetricsName: PollForCEA
+  Script: |
+    set -eou pipefail
+
+    getState() {
+      curl -s -S http://10.2.0.160:27182/api/v1/progress | jq '.progress.info' --raw-output
+    }
+
+    state=$(getState)
+    while [ "$state" != "change event application" ]
+    do
+      echo "Polling for change event application, current state=$state"
+      sleep 1;
+      state=$(getState)
+    done
+
+DrainWrites:
+  Repeat: 1
+  Command: sh
+  MetricsName: DrainWrites
+  Script: |
+    set -eou pipefail
+
+    getLag () {
```

Comment:
A lot of these scripts end up being pretty similar. I think it could be neat if we could come up with a way to add functions that can be re-used for external script actors in the future.

Reply:
Alternatively you are already rsyncing the replicator node. You can upload a bunch of arbitrary scripts and call them. You don't need to dynamically generate each of them. I would encourage you to not do that in shell script though. Node or python would be better.

Reply:
hmm that's a good point, I suppose we could just have a python script that lives alongside the workload, and give the external script actor a `ScriptPath` argument which could be provided in place of `Script`. Is that what you were thinking?

Reply:
yes, something like that.  It's all thinking out loud though.

Reply:
Sounds good! What are your thoughts on this proposal?

1. I'll change up some of the minor suggestions (e.g. cleanup, docs, etc.)
2. Merge this change without adding the EnvironmentVariables/ScriptPath/switch to python part
3. In the ticket to make this work with multiple mongosyncs, I'll implement 2) as I think it will be the most efficient path to get a correct, working solution

I think if we move forward before doing 2), that will let us shake out any surprises as we'll have this test running on a regular basis off of the waterfall over the next couple weeks

Reply:
That general plan sounds good. I'm not sure which thing you are proposing with "2)", but I don't think that matters right now for your next steps.